README.md

```diff
@@ -0,0 +1,45 @@
+OpenID Connect Relying Party
+============================
+
+[![Build Status](https://secure.travis-ci.org/GJWT/nodeOIDCClient?branch=master)](http://travis-ci.org/GJWT/nodeOIDCClient)
+
+Helper to handle client communication with an OpenID Connect OP.
+
+Note that for a Relying Party (RP) you would probably use the higher level API
+from https://secure.travis-ci.org/GJWT/nodeOIDCRP instead.
+
+This library is responsible for both OIDC discovery and registration.
+
+
+Example usage from an RP without prior client
+---------------------------------------------
+
+Getting an `identifier` from the user, using webfinger to discover the OP
+issuer, getting openid-configuration and using that to register a new client.
+
+```javascript
+const oidcclient = require('oidc-client');
+
+/**
+ * Get a client to use, either from storage or register a new one.
+ */
+async function getClient({ identifier, getClient, registerClient, clientMetadata }) {
```

Comment:
Could we use a different name for this function? As one of the destructured argument is also named getClient, it becomes confusing.

Reply:
Good catch.  Actually, eslint would have caught this and the other things, but it's not setup to check examples in a markdown file.  But I'll need to conclude the discussions with Roland about what belongs where first, then I'll update it.